components/brave_ads/core/internal/account/confirmations/redeem_unblinded_payment_tokens/redeem_unblinded_payment_tokens.cc

```diff
@@ -31,29 +31,46 @@ namespace {
 
 constexpr base::TimeDelta kRetryAfter = base::Minutes(1);
 
-constexpr base::TimeDelta kNextRedemptionAfter = base::Days(1);
-constexpr base::TimeDelta kDebugNextRedemptionAfter = base::Minutes(2);
-constexpr base::TimeDelta kNextRedemptionWhenExpiredAfter = base::Minutes(1);
+constexpr base::TimeDelta kNextTokenRedemptionAfter = base::Days(1);
+constexpr base::TimeDelta kDebugNextTokenRedemptionAfter = base::Minutes(2);
+constexpr base::TimeDelta kMinimumDelayBeforeRedeemingTokens = base::Minutes(1);
 
-base::TimeDelta CalculateRedemptionDelay() {
-  const base::Time next_token_redemption_at =
-      AdsClientHelper::GetInstance()->GetTimePref(
-          prefs::kNextTokenRedemptionAt);
+void SetNextTokenRedemptionAt(const base::Time next_token_redemption_at) {
+  AdsClientHelper::GetInstance()->SetTimePref(prefs::kNextTokenRedemptionAt,
+                                              next_token_redemption_at);
+}
+
+base::Time NextTokenRedemptionAt() {
+  return AdsClientHelper::GetInstance()->GetTimePref(
+      prefs::kNextTokenRedemptionAt);
+}
 
+bool HasPreviouslyRedeemedTokens() {
+  return !NextTokenRedemptionAt().is_null();
+}
+
+bool ShouldHaveRedeemedTokensInThePast() {
+  return NextTokenRedemptionAt() < base::Time::Now();
```

Comment:
I asked because the previous code used `>=`

Also, should we return `kMinimumDelayBeforeRedeemingTokens` if `NextTokenRedemptionAt() - base::Time::Now()` is less then `kMinimumDelayBeforeRedeemingTokens`?

Reply:
Yes we should based on our previous changes :-)

Reply:
Fixed